tests/test_gui.py

```diff
@@ -0,0 +1,1018 @@
+        """
+        :param root:
+        """
+        self.root = parent["Shotgun: Shotgun Toolkit Demos"].get()
+
+    def exists(self):
+        """
+        ``True`` if the widget was found, ``False`` otherwise.
+        """
+        return self.root.exists()
+
+    def open_demo_pane(self, name):
+        self.root["Demo Tree View"][name].get().mouseClick()
+
+    def close(self):
+        self.root.buttons["Close"].get().mouseClick()
+
+
+def test_activity_stream(app_dialog):
+    # Validate that Demo app is on the welcome page
+    assert app_dialog.root.captions[
+        "Help With this App"
+    ].exists(), "Not on the Demos app Welcome Page"
+
+    # Click on the Activity Stream widget
+    app_dialog.open_demo_pane("Activity Stream")
+    assert app_dialog.root.captions[
+        "Activity Stream"
+    ].exists(), "Not on the Activity Stream widget"
+
+    # Wait until note creation field is showing up.
+    while app_dialog.root.captions["Loading Shotgun Data..."].exists():
+        time.sleep(1)
+
+    # Click to create a new note
+    app_dialog.root.captions["Click to create a new note..."].get().mouseClick()
+
+    # Validate that all buttons are available
+    assert app_dialog.root.buttons[
+        "Cancel"
+    ].exists(), "Cancel buttons is not showing up"
+    assert app_dialog.root.buttons[
+        "Attach Files"
+    ].exists(), "Attach Screenshot buttons is not showing up"
+    assert app_dialog.root.buttons[
+        "Take Screenshot"
+    ].exists(), "Take Screenshot buttons is not showing up"
+    assert app_dialog.root.buttons[
+        "Create Note"
+    ].exists(), "Create Note buttons is not showing up"
+
+    # Add a note
+    app_dialog.root.textfields.typeIn("New Note")
+    app_dialog.root.buttons["Create Note"].get().mouseClick()
+    app_dialog.root.waitIdle(), 30
+    app_dialog.root.captions["New Note"].get().waitExist(), 30
+
+    # Validate the Note gets created
+    assert app_dialog.root.captions["New Note"].exists(), "New note wasn't created"
+    assert app_dialog.root.captions[
+        "Reply to this Note"
+    ].exists(), "New note wasn't created"
+
+    # Scroll down in the activity stream
+    activityScrollBar = first(app_dialog.root.scrollbars[1])
+    width, height = activityScrollBar.size
+    app_dialog.root.scrollbars[1]["Position"].get().mouseSlide()
+    activityScrollBar.mouseDrag(width * 0, height * 1)
+    assert app_dialog.root.buttons[
+        "Click here to see the Activity stream in Shotgun."
+    ].exists(), "Hyperlink to see the Activity Stream in Shotgun is missing"
+
+
+def test_context_selector(app_dialog):
+    # Click on the Context Selector widget
+    app_dialog.open_demo_pane("Context Selector Widget")
+    assert app_dialog.root.captions[
+        "Context Selector Widget"
+    ].exists(), "Not on the Context Selector Widget"
+
+    # Make sure you are in editing mode
+    if app_dialog.root.captions["Editing is now disabled."].exists():
```

Comment:
A comment explaining why we might not be in editing mode would be appreciated.

Reply:
It was just a check I added, but normally we should be in editing mode by default. Maybe I should remove that check in case there is a reel bug where the context selector widget editing mode is disabled by default?

Reply:
yeah, maybe remove the check and run the test a couple of times and see what happens?

Reply:
removed and run it 10 times in a row without any issue :)